### ISSUE TITLE
Dart support

### ISSUE THREAD

**Issue by dnfield** [github/semantic#461]
I've read through a couple issues and https://github.com/github/semantic/blob/master/docs/adding-new-languages.md - mainly filing this so it's tracked somewhere and people out there realize it's desired.

I'm a bit curious - is there any option to support talking to a language server directly?  Otherwise, each new language feature over time will take additional grammar updates/parser updates.  If it's somehow possible to just integrate with a langauge server, it'd be much easier to implement Dart support.

If it makes sense to file a separate bug for LSP support I'd be happy to.

**Comment by patrickt** [comment]
@dnfield LSP doesn’t provide enough information to be a proper driver for Semantic, unfortunately. The process of adding new languages is about to get much easier, so that documentation will (thankfully!) be out of date soon.

**Comment by dnfield** [comment]
Will it still be necessary to do the tree sitter parser?  As in, would it be worth starting on that with the upcoming changes?

**Comment by robrix** [comment]
@dnfield: Great question! Yes, definitely 👍

**Comment by trentgrover-wf** [comment]
any update on this?

**Comment by ChristianEdwardPadilla** [comment]
Did the process of adding a new language get easier? Maybe Dart can be revisited?

**Comment by patrickt** [comment]
@ChristianEdwardPadilla Yes! Take a look at the tree-sitter documentation: https://tree-sitter.github.io/tree-sitter/code-navigation-systems
